packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -35,99 +53,198 @@ contract DowntimeSlasher is SlasherUtil {
-   * @dev Due to getParentSealBitmap, startBlock must be within 4 epochs of the current head.
+   * @notice Calculates and returns the signature bitmap for the specified interval.
+   * This bitmap will contain a one for any validator that signed at least one block in that
+   * interval, and zero otherwise.
+   * @param startBlock First block of the interval.
+   * @param endBlock Last block of the interval.
+   * @return The signature uptime bitmap for the specified interval.
+   * @dev startBlock and endBlock must be in the same epoch.
+   * @dev The getParentSealBitmap precompile requires that startBlock must be within 4 epochs of 
+   * the current block.
    */
-  function isDown(uint256 startBlock, uint256 startSignerIndex, uint256 endSignerIndex)
+  function getBitmapForInterval(uint256 startBlock, uint256 endBlock)
     public
     view
-    returns (bool)
+    returns (bytes32)
   {
-    uint256 endBlock = getEndBlock(startBlock);
-    require(endBlock < block.number.sub(1), "end block must be smaller than current block");
+    require(endBlock >= startBlock, "endBlock must be greater or equal than startBlock");
+    // The signature bitmap for block N is stored in block N+1.
+    // The latest block is `block.number - 1`, which stores the signature bitmap for
+    // `block.number - 2`.
+    uint256 lastBlockWithSignatureBitmap = block.number.sub(2);
+    require(
+      endBlock <= lastBlockWithSignatureBitmap,
+      "the signature bitmap for endBlock is not yet available"
+    );
+    uint256 epochSize = getEpochSize();
     require(
-      startSignerIndex < numberValidatorsInSet(startBlock),
-      "Bad validator index at start block"
+      lastBlockWithSignatureBitmap.sub(startBlock) <= epochSize.mul(4),
+      "startBlock must be within 4 epochs of the current head"
     );
-    require(endSignerIndex < numberValidatorsInSet(endBlock), "Bad validator index at end block");
-    address startSigner = validatorSignerAddressFromSet(startSignerIndex, startBlock);
-    address endSigner = validatorSignerAddressFromSet(endSignerIndex, endBlock);
-    IAccounts accounts = getAccounts();
     require(
-      accounts.signerToAccount(startSigner) == accounts.signerToAccount(endSigner),
-      "Signers do not match"
+      epochNumberOfBlock(startBlock, epochSize) == epochNumberOfBlock(endBlock, epochSize),
+      "startBlock and endBlock must be in the same epoch"
     );
-    uint256 sz = getEpochSize();
-    uint256 startEpoch = epochNumberOfBlock(startBlock, sz);
-    for (uint256 n = startBlock; n <= endBlock; n = n.add(1)) {
-      uint256 signerIndex = epochNumberOfBlock(n, sz) == startEpoch
-        ? startSignerIndex
-        : endSignerIndex;
-      // We want to check signers for block n,
-      // so we get the parent seal bitmap for the next block
-      if (uint256(getParentSealBitmap(n.add(1))) & (1 << signerIndex) != 0) return false;
+
+    bytes32 bitmap;
+    for (uint256 n = startBlock; n <= endBlock; n++) {
```

Comment:
Naming suggestion: `n` -> `block`.

Reply:
change it to `blockNumber` to avoid confusions with `block` global